docs/serving/samples/cloudevents/cloudevents-vertx/src/main/java/org/knative/examples/cloudevents/vertx/CloudEventReceiverVerticle.java

```diff
@@ -0,0 +1,43 @@
+package org.knative.examples.cloudevents.vertx;
+
+import io.cloudevents.http.vertx.VertxMessageFactory;
+import io.cloudevents.CloudEvent;
+import io.cloudevents.message.Message;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Vertx;
+
+public class CloudEventReceiverVerticle extends AbstractVerticle {
+
+  public static void main(String[] args) {
+    Vertx vertx = Vertx.vertx();
+    vertx.deployVerticle(new CloudEventReceiverVerticle());
+  }
+
+  public void start() {
+    vertx.createHttpServer()
+        .requestHandler(req -> {
+          // Transform the HttpRequest to Event
+          VertxMessageFactory.fromHttpServerRequest(req)
+              .map(Message::toEvent)
+              .onComplete(asyncResult -> {
+                if (asyncResult.succeeded()) {
+                  CloudEvent event = asyncResult.result();
+
+                  System.out.println("Received event: " + event);
+                  req.response().setStatusCode(202).end();
```

Comment:
I've been trying to get folks to align quite literally with `cloudevents-go` because it illustrates a very useful cross-section of functionality per SDK.
1. If `K_SINK` is set, then it sends a result event there (useful for source authors)
2. If it isn't set, then it responds with the result event (useful for interacting with Broker).

`2.` is especially useful because the SDKs upstream don't have great samples that *return* cloud events, which is fundamental to our broker model (which is partially why I started this flavor of sample).

`1.` is designed to make this compose really nicely with SinkBinding (the sink binding sample cross-links to these samples!).

So there's some method to my madness, sorry if it seems pedantic, but hopefully it makes sense. 😅

Reply:
Ok give me a couple of days and i'll provide it :smile:

Reply:
Good points, @mattmoor